### DISCUSSION TITLE
v0.3.1 Released

### DISCUSSION THREAD

**Discussion by tomvothecoder** [xCDAT/xcdat#322] [1. Announcements]
## v0.3.1 (18 August 2022)

This patch release focuses on bug fixes including handling bounds generation with singleton coordinates and the use of ``cftime``
to represent temporal averaging outputs and non-CF compliant time coordinates (to avoid the pandas Timestamp limitations).

#### Bounds
* Ignore singleton coordinates without dims when attempting to generate bounds [Stephen Po-Chedley](https://github.com/pochedls) in https://github.com/xCDAT/xcdat/pull/281
* Modify logic to not throw error for singleton coordinates (with no bounds) by [Stephen Po-Chedley](https://github.com/pochedls) in https://github.com/xCDAT/xcdat/pull/313

#### Time Axis and Coordinates
* Fix `TypeError` with Dask Arrays from multifile datasets in temporal averaging by [Stephen Po-Chedley](https://github.com/pochedls) in https://github.com/xCDAT/xcdat/pull/291
* Use `cftime` to avoid out of bounds `datetime` when decoding non-CF time coordinates by [Stephen Po-Chedley](https://github.com/pochedls) and [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/283
* Use `cftime` for temporal averaging operations to avoid out of bounds `datetime` by [Stephen Po-Chedley](https://github.com/pochedls) and [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/302
* Fix `open_mfdataset()` dropping time encoding attrs by [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/309
* Replace "time" references with `self._dim` in `class TemporalAccessor` by [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/312

### Internal Changes
* Filters safe warnings. by [Jason Boutte](https://github.com/jasonb5) in https://github.com/xCDAT/xcdat/pull/276

### Documentation
* update conda install to conda create by [Paul Durack](https://github.com/durack1) in https://github.com/xCDAT/xcdat/pull/294
* Update project overview and planned features list by [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/298
* Fix bullet formatting in `README.rst` and`index.rst` by [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/299
* Fix Jupyter headings not rendering with pandoc by [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/318

### DevOps
* Unify workspace settings with `settings.json` by [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/297
* Run CI/CD on "push" and "workflow_dispatch" by [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/287 and https://github.com/xCDAT/xcdat/pull/288
* Pin `numba=0.55.2` in dev env and constrain `numba>=0.55.2` in ci env by [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/280
* Update conda env yml files and add missing dependencies by [Tom Vo](https://github.com/tomvothecoder) in https://github.com/xCDAT/xcdat/pull/307

### New Contributors
* [Paul Durack](https://github.com/durack1) made their first contribution in https://github.com/xCDAT/xcdat/pull/294

**Full Changelog**: https://github.com/xCDAT/xcdat/compare/v0.3.0...v0.3.1

<hr /><em>This discussion was created from the release <a href='https://github.com/xCDAT/xcdat/releases/tag/v0.3.1'>v0.3.1</a>.</em>

**Comment by tomvothecoder** [comment]
Hi @xCDAT/core-developers, @durack1, @msahn, @jypeter, @oliviermarti, @gleckler1, @crterai,

xCDAT v0.3.1 has been released today in conjunction with our current testing phase! 

This patch release focuses on bug fixes including handling bounds generation with singleton coordinates and the use of `cftime` to represent temporal averaging outputs and non-CF compliant time coordinates (to avoid the pandas Timestamp limitations).

Please view the below links if you are interested in learning more.

* Changelog: https://xcdat.readthedocs.io/en/v0.3.1/history.html#v0-3-1-18-august-2022
* Documentation: https://xcdat.readthedocs.io/en/v0.3.1/index.html
* Anaconda page: https://anaconda.org/conda-forge/xcdat

We are now practicing rapid release cycles, which involves releasing new versions based on the significance of bug fixes and features.

Next milestones:
*	Release v0.3.2 with additional bug fixes 
   * v0.3.2 Project Board: https://github.com/orgs/xCDAT/projects/5?query=is%3Aopen+sort%3Aupdated-desc
*	Wrap up the testing phase in a month or so
*	Explore vertical regridding solutions

**Reply by pochedls** [reply]
Thanks @tomvothecoder - I realize these releases take time. These bug fixes are really helpful!

**Reply by tomvothecoder** [reply]
No problem @pochedls. Most of the time is spent with updating the conda env dependencies and/or updating up the changelog.
The actual time to release on GH and conda-forge is pretty minimal.

I might make it a routine to update conda env deps only for minor or major releases.
